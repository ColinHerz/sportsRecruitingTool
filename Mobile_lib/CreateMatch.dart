import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;
import 'package:sporta/ShowDialog.dart';
import 'models/GetBags.dart';
import 'models/EventInfo.dart';
import 'Session.dart';
import 'dart:convert';
import 'dart:async';

class CreateMatch extends StatefulWidget{
  _CreateMatch createState() => _CreateMatch();
}

class _CreateMatch extends State<CreateMatch>{
  final _matchNameController = TextEditingController();
  final _courseController = TextEditingController();
  List<TextEditingController> _parsController;
  List<TextEditingController> _controllerList;
  String _currentBag = '[None selected]';
  List<String> _bagIDlist = new List<String>();
  List<String> _bagList = new List<String>();
  DateTime selectedDate = DateTime.now();

  Future<GetBags> getBags() async {
    List<dynamic> bagResponseList;
    final http.Response _response = await http.get(
      'http://localhost3000.us-east-2.elasticbeanstalk.com/api/golf/getAllGolfBags',
      headers: headers,
    );

    if (_response.statusCode == 200) {
      _bagList.add('[None selected]');
      _bagIDlist.add('null');
      bagResponseList = json.decode(_response.body);
      setState(() {
        for (int i = 0; i < bagResponseList.length; i++) {
          _bagList.add(bagResponseList[i]['bagName']);
          _bagIDlist.add(bagResponseList[i]['_id']);
        }
      });
    }
    else {
      print(_response.statusCode);
      print(_response.body);
    }
  }
  Future<GetBags> createMatch(dynamic info) async {
    final http.Response _response = await http.post(
      'http://localhost3000.us-east-2.elasticbeanstalk.com/api/golf/createGolfMatch',
      headers: headers,
      body: info
    );

    if (_response.statusCode == 200) {
    }
    else {
      print(_response.statusCode);
      print(_response.body);
    }
  }
  void initState(){
    super.initState();
    getBags();
  }

  Future<void> _selectDate(BuildContext context) async {
    final DateTime picked = await showDatePicker(
        context: context,
        initialDate: selectedDate,
        firstDate: DateTime(2015, 8),
        lastDate: DateTime(2101));
    if (picked != null && picked != selectedDate)
      setState(() {
        selectedDate = picked;
      });
  }

  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Create Match'),
      ),
      body: Center(
        child: ListView(
          shrinkWrap: true,
          children: <Widget>[
            Padding(
                padding: EdgeInsets.fromLTRB(0, 0, 0, 10),
                child: TextField(
                    controller: _matchNameController,
                    decoration: InputDecoration(
                      border: OutlineInputBorder(),
                      labelText: 'Match Name',
                    )
                )
            ),
            Padding(
                padding: EdgeInsets.fromLTRB(0, 0, 0, 10),
                child: TextField(
                    controller: _courseController,
                    decoration: InputDecoration(
                      border: OutlineInputBorder(),
                      labelText: 'Course',
                    )
                )
            ),
            Text("${selectedDate.toLocal()}".split(' ')[0],
                textAlign: TextAlign.center,
              style: TextStyle(fontSize: 20, fontWeight: FontWeight.bold),
            ),
            Align(
              child: RaisedButton(
                onPressed: (){
                  _selectDate(context);
                },
                child: Text("Select Date"),
              ),
            ),
            Padding(
              padding: EdgeInsets.fromLTRB(0, 10, 0, 10),
            ),
            Text("Select bag:",
                textAlign: TextAlign.center,
                style: TextStyle(
                    fontSize: 20,
                    fontWeight: FontWeight.bold)
            ),
            Center(
              child: DropdownButton<String>(
                value: _currentBag,
                icon: Icon(Icons.arrow_downward),
                iconSize: 24,
                elevation: 16,
                style: TextStyle(color:Colors.deepOrange),
                onChanged: (String newValue) {
                  setState(() {
                    _currentBag = newValue;
                  });
                },
                items: _bagList.map<DropdownMenuItem<String>>((String value) {
                  return DropdownMenuItem<String>(
                    value: value,
                    child: Text(value, style: TextStyle(fontSize: 18)),
                  );
                }).toList(),
              ),
            ),
            Align(
              child: RaisedButton(
                  onPressed: (){
                    if (_matchNameController.text == "" || _courseController.text == "" || _currentBag == '[None selected]')
                      DialogPopUp(context, "Please fill in all fields");
                    else{
                      String bagID;
                      for (int i = 0; i < _bagList.length; i++){
                        if (_bagList[i] == _currentBag){
                          bagID = _bagIDlist[i];
                          break;
                        }
                      }
                      createMatch(
                        jsonEncode(<String, String>{
                          'nameOfRound': _matchNameController.text,
                          'coursePlayed': _courseController.text,
                          'datePlayed' : selectedDate.toIso8601String() + 'Z',
                          'GolfBagUsed' : bagID
                        })
                      );
                      Navigator.pop(context);
                    }
                  },
                  child: Text('Create Match')
              ),
            )
          ],
        ),
      ),
    );
  }
}
