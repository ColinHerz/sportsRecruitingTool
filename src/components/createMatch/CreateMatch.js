import apiCall from "../../api/api.js";
import { Link } from "react-router-dom";
import React, { useEffect, useState } from "react";
import { useForm } from "react-hook-form";

import "./createMatch.scss";

const CreateMatch = props => {
	const { register, handleSubmit, errors } = useForm();

	const [firstLoad, setFirstLoad] = useState(true);
	const [showError, setShowError] = useState(false);

	const [matchId, setMatchId] = useState(``);

	const [golfBags, setGolfBags] = useState([]);

	useEffect(() => {
		if (firstLoad) {
			apiCall(
				{
					endpoint: `/golf/getAllGolfBags`,
					type: `GET`
				},
				data => {
					if (data.status !== 200) {
						setShowError(true);
						return;
					}

					setGolfBags(JSON.parse(data.response));
				},
				() => {
					setShowError(true);
					return;
				}
			);

			setFirstLoad(false);
		}
	}, [firstLoad]);

	const onSubmit = info => {
		apiCall(
			{
				endpoint: `/golf/createGolfMatch`,
				type: `POST`,
				body: info
			},
			data => {
				if (data.status !== 200) {
					setShowError(true);
				}

				setMatchId(JSON.parse(data.response).golf);
			},
			() => {
				setShowError(true);
			}
		);
	};

	return (
		<main id="create-match">
			<h2>Create a Match</h2>

			{
				showError ?
					<p id="error-message">Something went wrong. Please try again later.</p>:
					null
			}

			{
				matchId === `` ?
					<form onSubmit={handleSubmit(onSubmit)}>
						<label>
							Match Name
							<input
								type="text"
								name="nameOfRound"
								ref={
									register({
										required: true
									})
								}
							/>
						</label>
						{errors.nameOfRound && <p className="error-text">A match name is required.</p>}

						<label>
							Course Played
							<input
								type="text"
								name="coursePlayed"
								ref={
									register({
										required: true
									})
								}
							/>
						</label>
						{errors.coursePlayed && <p className="error-text">A course name is required.</p>}

						<label>
							Date Played
							<input
								type="date"
								name="datePlayed"
								ref={
									register({
										required: true
									})
								}
							/>
						</label>
						{errors.datePlayed && <p className="error-text">A play date is required.</p>}

						<label>
							Bag Used
							<select
								name="GolfBagUsed"
								ref={
									register({
										required: true
									})
								}
							>
								<option disabled>Please select a bag</option>
								{
									golfBags.map(bag => (
										<option
											key={bag._id}
											value={bag._id}
										>
											{bag.bagName}
										</option>
									))
								}
							</select>
						</label>
						{errors.GolfBagUsed && <p className="error-text">A valid bag is required.</p>}

						<input type="submit" value="Create" />
					</form>:
					<React.Fragment>
						<p>Match created successfully.</p>

						<Link to={`/match/${matchId}/`}>View the match.</Link>
					</React.Fragment>
			}
		</main>
	);
};

export default CreateMatch;
