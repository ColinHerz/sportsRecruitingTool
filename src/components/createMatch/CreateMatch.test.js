/* eslint-env jest */
import React from "react";
import renderer from "react-test-renderer";

import CreateMatch from "./CreateMatch.js";

describe(`Rendering`, () => {
	it(`Renders the basic page correctly`, () => {
		const component = renderer.create(<CreateMatch />);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
