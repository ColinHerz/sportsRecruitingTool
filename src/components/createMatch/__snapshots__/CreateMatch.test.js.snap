// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Renders the basic page correctly 1`] = `
<main
  id="create-match"
>
  <h2>
    Create a Match
  </h2>
  <form
    onSubmit={[Function]}
  >
    <label>
      Match Name:
      <input
        name="nameOfRound"
        type="text"
      />
    </label>
    <label>
      Course Played:
      <input
        name="coursePlayed"
        type="text"
      />
    </label>
    <label>
      Date Played:
      <input
        name="datePlayed"
        type="date"
      />
    </label>
    <label>
      Bag Used:
      <select
        name="GolfBagUsed"
      >
        <option
          disabled={true}
        >
          Please select a bag
        </option>
      </select>
    </label>
    <input
      type="submit"
      value="Create"
    />
  </form>
</main>
`;
